packages/hop-node/src/watchers/CommitTransfersWatcher.ts

```diff
@@ -78,26 +79,23 @@ class CommitTransfersWatcher extends BaseWatcher {
   }
 
   async checkTransferSentFromDb () {
-    const dbTransfers = await this.db.transfers.getUncommittedTransfers({
-      sourceChainId: await this.bridge.getChainId()
-    })
+    const dbTransfers = await this.db.transfers.getUncommittedTransfers(await this.getFilterRoute())
     if (!dbTransfers.length) {
       return
     }
 
     this.logger.info(
         `checking ${dbTransfers.length} uncommitted transfers db items`
     )
-    const destinationChainIds: number[] = []
+
+    const seen: Record<number, boolean> = {}
     for (const dbTransfer of dbTransfers) {
       const { destinationChainId } = dbTransfer
-      if (!destinationChainIds.includes(destinationChainId!)) { // eslint-disable-line @typescript-eslint/no-non-null-assertion
-        destinationChainIds.push(destinationChainId!) // eslint-disable-line @typescript-eslint/no-non-null-assertion
+      if (!seen[destinationChainId!]) { // eslint-disable-line @typescript-eslint/no-non-null-assertion
+        seen[destinationChainId!] = true // eslint-disable-line @typescript-eslint/no-non-null-assertion
```

Comment:
If `this.checkIfShouldCommit()` fails for some reason, then I think this logic prevents it from being retried again. What is the purpose of this `seen` logic?

Reply:
I was trying to clean it up by using a map and one loop but I see what you mean now. 

Changes reverted in https://github.com/hop-protocol/hop/commit/4d5b5a96e8c1b000bda20ca710aaeaaab5ce568e